backend/src/bundles/talent-badges/talent-badge.repository.ts

```diff
@@ -0,0 +1,69 @@
+import { ErrorMessages } from '~/common/enums/enums.js';
+import { type Repository } from '~/common/types/repository.type.js';
+
+import { TalentBadgeEntity } from './talent-badge.entity.js';
+import { type TalentBadgeModel } from './talent-badge.model.js';
+import {
+    type TalentBadgeCreate,
+    type TalentBadgePatchAndFetch,
+} from './types/types.js';
+
+class TalentBadgeRepository implements Repository {
+    private talentBadgeModel: typeof TalentBadgeModel;
+
+    public constructor(talentBadgeModel: typeof TalentBadgeModel) {
+        this.talentBadgeModel = talentBadgeModel;
+    }
+
+    public findAll(): Promise<TalentBadgeEntity[]> {
+        throw new Error(ErrorMessages.NOT_IMPLEMENTED);
+    }
+
+    public async create({
+        badgeId,
+        userId,
+    }: TalentBadgeCreate): Promise<TalentBadgeEntity> {
+        const item = await this.talentBadgeModel
+            .query()
+            .insert({
+                badgeId,
+                isShown: true,
+                userId,
+            })
+            .returning('*')
+            .execute();
+
+        return TalentBadgeEntity.initializeNew(item);
+    }
+
+    public async find(
+        payload: Record<string, unknown>,
+    ): Promise<TalentBadgeModel | null> {
+        const badge = await this.talentBadgeModel
+            .query()
+            .findOne({ ...payload });
+
+        return badge ?? null;
+    }
+
+    public async update({
+        id,
+        isShown,
+        userDetailsId,
+    }: TalentBadgePatchAndFetch): Promise<TalentBadgeEntity> {
+        const talentBadge = await this.talentBadgeModel
+            .query()
+            .patchAndFetchById(id, {
+                isShown: !isShown,
```

Comment:
```suggestion
                isShown,
```
Why did you reverse it?

Reply:
In the second step we have only checkboxes and as I understand we should just reverse isShown of this badge if we already have it for the user in DB.

Reply:
Backend part should make sense, and that logic should be in the frontend part.

Reply:
as I undersood @MaksymGavatuk we should recieve from frontend only array of strings = Ids also without isShown and other fields

Reply:
Yes it just string[]

Reply:
Yes, but in this case, we need to set isShow as we provide from params

Reply:
done